src/coreclr/src/jit/lclvars.cpp

```diff
@@ -1829,6 +1848,12 @@ bool Compiler::StructPromotionHelper::CanPromoteStructVar(unsigned lclNum)
         return false;
     }
 
+    if (compiler->lvaIsOSRLocal(lclNum))
+    {
+        JITDUMP("  struct promotion of V%02u is disabled because it is an OSR local\n", lclNum);
```

Comment:
This seems like a pretty significant pessimization that would result in the OSR method potentially being considerably slower than the Tier1 method - am I missing something?

Reply:
I'm hoping it is just a workaround -- I need to think more about about the conditions that will allow safe struct promotion in OSR methods. I had promotion enabled for most of my prototyping but when running pri1 tests I hit a few test failures I could not sort through quickly.

One issue is that since we partially import the IL for OSR methods, we may miss seeing key information that should block promotion or force dependent promotion; this information needs to be produced during the Tier0 compile and conveyed via the patchpoint info. The current "address exposed" bit in the patchpoint info (which is based on `lvHasLdAddrOp`) is evidently not sufficient.

Reply:
Mentioned this more clearly in the doc; once we get this PR merged I'll open up a follow-on issue tracking the various workarounds that need re-examination.